src/main/java/mate/academy/bookshop/service/OrderServiceImpl.java

```diff
@@ -0,0 +1,123 @@
+package mate.academy.bookshop.service;
+
+import jakarta.transaction.Transactional;
+import java.math.BigDecimal;
+import java.time.LocalDateTime;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import lombok.RequiredArgsConstructor;
+import mate.academy.bookshop.dto.order.OrderDto;
+import mate.academy.bookshop.dto.order.OrderItemDto;
+import mate.academy.bookshop.dto.order.PlaceOrderRequestDto;
+import mate.academy.bookshop.dto.order.UpdateStatusRequestDto;
+import mate.academy.bookshop.exception.EntityNotFoundException;
+import mate.academy.bookshop.mapper.OrderItemMapper;
+import mate.academy.bookshop.mapper.OrderMapper;
+import mate.academy.bookshop.model.Book;
+import mate.academy.bookshop.model.CartItem;
+import mate.academy.bookshop.model.Order;
+import mate.academy.bookshop.model.OrderItem;
+import mate.academy.bookshop.model.ShoppingCart;
+import mate.academy.bookshop.model.User;
+import mate.academy.bookshop.repository.order.OrderRepository;
+import mate.academy.bookshop.repository.orderitem.OrderItemRepository;
+import mate.academy.bookshop.repository.shoppingcart.ShoppingCartRepository;
+import mate.academy.bookshop.repository.user.UserRepository;
+import org.springframework.data.domain.Pageable;
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+public class OrderServiceImpl implements OrderService {
+    private final ShoppingCartRepository shoppingCartRepository;
+    private final OrderRepository orderRepository;
+    private final UserRepository userRepository;
+    private final OrderItemRepository orderItemRepository;
+    private final OrderMapper orderMapper;
+    private final OrderItemMapper orderItemMapper;
+
+    @Override
+    @Transactional
+    public OrderDto placeOrder(Long userId, PlaceOrderRequestDto requestDto) {
```

Comment:
Can you simplify this method? (By extracting some logical parts into separate private methods for example)